### PR TITLE
Support and test Public Access Prevention

### LINKED ISSUE #5672: storage: add support for PublicAccessPrevention
We need to add support for a new field publicAccessPrevention in the bucket IAM configuration for storage. Full details are available in [this doc](https://docs.google.com/document/d/1o_ElQg4c8enEx8OYa3irrUYFKADmHFbRn7dC9g287sI/). This should include ability to create buckets with this field set, get the field, and set the field. We'd also like an integration test that verifies that the feature works as expected.

Please wait for approval from myself or @shaffeeullah before merging or releasing this change so we can coordinate around release dates.

I'm not totally sure to what degree the code will need to be changed in this library (e.g. https://github.com/googleapis/google-cloud-dotnet/blob/1b58c22a34ed0da5b65bc51da5b35d9314dbcfc8/apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/SetBucketIamPolicyOptions.cs doesn't seem to reference specific options in the code) but there should definitely be an integration test along the lines of what's present for UBLA. The field has already been published in the storage discovery doc.

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #5672

### CODE REVIEW

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/PublicAccessPreventionTest.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            var client = _fixture.Client;
+            string bucketName = _fixture.GenerateBucketName();
+            Bucket bucket = _fixture.CreateBucket(bucketName, false);
+            Assert.Null(bucket.IamConfiguration.PublicAccessPrevention);
+
+            // Enforce PAP
+            client.PatchBucket(CreateBucketRepresentation(bucketName, EnforcedValue));
+
+            // We shouldn't be able to all users to view objects.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/PublicAccessPreventionTest.cs:46]: nit: Incomplete comment.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/PublicAccessPreventionTest.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            var client = _fixture.Client;
+            string bucketName = _fixture.GenerateBucketName();
+            StorageFixture.SleepAfterBucketCreateDelete();
+            Bucket created = client.CreateBucket(_fixture.ProjectId, CreateBucketRepresentation(bucketName, EnforcedValue));
+            _fixture.RegisterBucketToDelete(bucketName);
+            StorageFixture.SleepAfterBucketCreateDelete();
+
+            Assert.Equal(EnforcedValue, created.IamConfiguration.PublicAccessPrevention);
+
+            // We shouldn't be able to all users to view objects.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/PublicAccessPreventionTest.cs:65]: nit: Incomplete comment.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/PublicAccessPreventionTest.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/PublicAccessPreventionTest.cs:46]: Whoops, thanks. Will just add "allow" after the "to"

**Pull request merged**